Modules/Scripted/DataProbe/DataProbe.py

```diff
@@ -328,7 +328,21 @@ def generateLayerName(self, slicerLayerLogic):
 
     def generateIJKPixelDescription(self, ijk, slicerLayerLogic):
         volumeNode = slicerLayerLogic.GetVolumeNode()
-        return f"({ijk[0]:3d}, {ijk[1]:3d}, {ijk[2]:3d})" if volumeNode else ""
+        if not volumeNode:
+            return ""
+        description = f"({ijk[0]:3d}, {ijk[1]:3d}, {ijk[2]:3d})"
+        instanceNumbersAttr = volumeNode.GetAttribute("DICOM.instanceNumbers")
+        if instanceNumbersAttr:
+            try:
+                # The DICOMScalarVolumePlugin ensures that the instance numbers are sorted
+                # so that the i-th slice in the volume node corresponds to the i-th value in the DICOM.instanceNumbers attribute.
+                instanceNumbers = [int(x) for x in instanceNumbersAttr.split(" ")]
+                k = ijk[2]
+                if 0 <= k < len(instanceNumbers):
+                    description += f" [dcm: {instanceNumbers[k]}]"
```

Comment:
Make the string translatable. Append direction should not be hardcoded with `+=` and allow translators to change the punctuation and order.

```suggestion
                    description = _("{description} [slice: {instanceNumber}]").format(description=description, instanceNumber=instanceNumbers[k]```